tests/test_downloadermiddleware_httpcache.py

```diff
@@ -156,7 +157,11 @@ def _get_settings(self, **new_settings):
 
 class LeveldbStorageTest(DefaultStorageTest):
 
-    pytest.importorskip('leveldb')
+    try:
+        pytest.importorskip('leveldb')
+    except SystemError:
+        # Happens in python 3.8
+        pytestmark = pytest.skip("'SystemError: bad call flags' occurs on Python 3.8")
```

Comment:
Is the assignment needed?

```suggestion
        pytest.skip("'SystemError: bad call flags' occurs on Python 3.8")
```

Reply:
Yes - travis CI called it out specifically in the last build https://travis-ci.org/scrapy/scrapy/jobs/603880139?utm_medium=notification&utm_source=github_status

```
Using pytest.skip outside of a test is not allowed. To decorate a test function, use the @pytest.mark.skip or @pytest.mark.skipif decorators instead, and to skip a module use `pytestmark = pytest.mark.{skip,skipif}.
```

Reply:
This is getting complex.

I think you may be able to move the `try-except` to something like https://docs.pytest.org/en/latest/xunit_setup.html#method-and-function-level-setup-teardown, but feel free to go back to your initial approach of decorators based on the running Python version.

Reply:
I'm not sure how using a setup method is gonna make it less complex? You mean making something like or is there another approach I'm missing?

```
class LeveldbStorageTest(DefaultStorageTest):
    storage_class = 'scrapy.extensions.httpcache.LeveldbCacheStorage'

    @classmethod
    def setup_class(cls, *args, **kwargs):
        test_class = cls(*args, **kwargs)
        try:
            test_class.leveldb = pytest.importorskip('leveldb')
        except SystemError:
            test_class.pytestmark = pytest.mark.skip("'SystemError: bad call flags' occurs on Python >= 3.8")

        return test_class
```

Right now I have it refactored to a single check localized to the specific method where the problem arises, with clear commenting/logging when it triggers. Reverting back to the multiple decorators or incorporating a setup method seems like it'd be adding complexity as I am adding adding checks to multiple parts of the code or I am having to incorporate additional syntax to mimic the current behavior.